aspen/website.py

```diff
@@ -68,15 +69,19 @@ def redirect(self, location, code=None, permanent=False, base_url=None, response
         permanent is True and 302 (Found) if it is False. If url doesn't start
         with base_url (defaulting to self.base_url), then we prefix it with
         base_url before redirecting. This is a protection against open
-        redirects. If you provide your own response we will set .code and
-        .headers['Location'] on it.
+        redirects. If base_url is not '' and location is not an absolute path,
```

Comment:
Ah! So then how about "base_url must be empty if you wish to use a relative path as location; if it's not, a BadLocation error will be thrown." ?  It's much clearer, IMO, than "If base_url is not '' and location is not an absolute path, we raise BadLocation"